src/model.ts

```diff
@@ -1,13 +1,30 @@
 import { Request } from 'express';
-import { Readable } from 'stream';
+import { PagingStream } from './paging-stream';
+import { ISearchApiOptions } from './types';
+import { getSearchQueryParams } from './get-search-query-params';
 
 export class HubApiModel {
-  async getStream (request: Request) {
-    console.log(request);
+  private _options: any;
 
-    const ret = new Readable({ read () {} });
-    ret.push({ foo: 'bar' });
-    ret.push(null);
-    return ret;
+  constructor (options: any) {
+    this._options = options;
+  }
+
+  getStream (request: Request) {
+    // TODO revisit how we get these search options
+    const searchOptions: ISearchApiOptions = request.app.locals.searchOptions;
+    const siteUrl = request.app.locals.hubSiteUrl || this._options.defaultSiteUrl;
+
+    const searchApiStream = new PagingStream({
+      firstPage: `${siteUrl}/api/v3/datasets?${getSearchQueryParams(searchOptions)}`,
+
+      loadPage: (url: string) => fetch(url).then(res => res.json()),
+
+      streamPage: (response, push) => response.data.forEach(dataset => push(dataset)),
+
+      getNextPage: response => response.links.next
```

Comment:
I think the name here is possibly confusing.  It looks like this function gets the parameters needed for loading the next page, but doesn't get the next page itself.  I think `getNextPageParams` would be more descriptive.